src/libraries/Microsoft.Extensions.Hosting/src/HostBuilder.cs

```diff
@@ -214,7 +216,14 @@ private void CreateServiceProvider()
 #pragma warning restore CS0618 // Type or member is obsolete
             services.AddSingleton<IHostApplicationLifetime, ApplicationLifetime>();
             services.AddSingleton<IHostLifetime, ConsoleLifetime>();
-            services.AddSingleton<IHost, Internal.Host>();
+            services.AddSingleton<IHost>(serviceProvider=>
+            {
+                return new Internal.Host(_appServices,
+                    serviceProvider.GetRequiredService<IHostApplicationLifetime>(),
```

Comment:
would serviceProvider and _appServices be the same then?

Reply:
Yes I don't like that this code is using different service providers.

Reply:
At this lines serviceProvider (ServiceProviderEngineScope) != _appServices (ServiceProvider).
The point of PR that `Host` will have ref to the same root IServiceProvider (_appServices) as in `HostBuilder`